### PR TITLE
PYIC-6652: remove address/fraud + gpg45 evaluation from reverification journey

### PR DESCRIPTION
**Pull request by thebauSoftwire**
### What changed
- removed address/fraud + gpg45 evaluation from reverification journey
- in `CriStoringService.storeVcs`, we only VCs against a user if we are not on a reverification journey as during this journey, they are not authenticated
- in `CriCheckingService.checkVcResponse`, we only want to check CIs when not on a reverification journey
- adds reverificationStatus to ipvSession which is set in `CheckCoiHandler`
- updates `UserReverificationHandler ` to check for reverificationStatus instead of Vot
- updates `gpg45DcmawValidator` to also check verification and strength scores
- adds/updates unit tests

### Why did it change
We are removing the address and fraud checks from the MFA reset journey, and relying on the verification check in DCMAW and continuity of identity check only.

### Issue tracking

- [PYIC-6652](https://govukverify.atlassian.net/browse/PYIC-6652)

[PYIC-6652]: https://govukverify.atlassian.net/browse/PYIC-6652?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy/template.yaml
Comment (Joe-Edwards-GDS) [inline deploy/template.yaml:2891]: (We could potentially merge this and the one above into a CRUD policy - but this is fine)

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/ClientOAuthSessionItem.java
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/ClientOAuthSessionItem.java]: Do you think it'd make sense to have this return a `List`? (Is an array ever more useful?)

File: lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java
Comment (Joe-Edwards-GDS) [inline lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java]: It might be slightly cleaner to make this a single call above the `if (!coiCheckSuccess)` - but doesn't matter too much.

(Just so if other branches are added, it's not missed)

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/ReverificationStatus.java
Comment (sam803) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/ReverificationStatus.java]: Is this status string used anywhere? If not simpler to remove and just have:
```
public enum ReverificationStatus {
    SUCCESS,  FAILED;
}
```

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingServiceTest.java
Comment (Joe-Edwards-GDS) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingServiceTest.java]: (I think this comment is wrong?)

File: libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java
Comment (Joe-Edwards-GDS) [inline libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java]: Do you think there's anywhere to commonise this? Perhaps it should be a method on `CredentialEvidenceItem` itself.

I might just override `getVerificationScore` itself so it follows this logic for DCMAW.

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/ReverificationStatus.java
Comment (Joe-Edwards-GDS) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/ReverificationStatus.java]: TBH it could also just be a boolean (although an enum does give us more flexibility to extend in future)

File: deploy/template.yaml
Comment (thebauSoftwire) [reply deploy/template.yaml:2891]: yeah that was the other option but I was leaning towards this in keeping w least privilege

File: libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java
Comment (thebauSoftwire) [reply libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java]: Not sure it should be in `CredentialEvidenceItem` as I don't think it's a method that should be available if it wasn't from dcmaw. But I agree with replacing `getVerificationScore` instead and letting the validator be the source of truth for this logic. Let me know if you disagree though!

File: lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java
Comment (Joe-Edwards-GDS) [inline lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java:158]: I really meant make this cover all cases for the check, i.e.:
```
if (isReverification) {
    setIpvSessionReverificationStatus(ipvSession, coiCheckSuccess
        ? ReverificationStatus.SUCCESS
        : ReverificationStatus.FAILED);
}
```
(We definitely don't want to do two writes to dynamodb if we don't need to)

File: lambdas/check-coi/src/test/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandlerTest.java
Comment (Joe-Edwards-GDS) [inline lambdas/check-coi/src/test/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandlerTest.java]: (Personally I have a slight preference for `List.of()` - but it's fairly equivalent)

File: libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java
Comment (Joe-Edwards-GDS) [reply libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java]: You could still include the DCMAW check if it was part of `CredentialEvidenceItem`, e.g.:
```
public Integer getVerificationScore() {
    if (isDcmaw()) {
        // get biometric verification score
    }
    return verificationScore;
}
```
but either way is fine

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Ah, looks like you've got conflicts from Chris' merge 😬

File: libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java
Comment (thebauSoftwire) [reply libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45DcmawValidator.java]: good point!

File: libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItem.java
Comment (Joe-Edwards-GDS) [inline libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItem.java]: This is fine, but if you want a couple of neater ways of writing this:

Streams:
```
return checkDetails.stream()
        .filter(checkDetail -> checkDetail.getBiometricVerificationProcessLevel() != null)
        .findFirst()
        .map(CheckDetail::getBiometricVerificationProcessLevel)
        .orElse(0);
```

Or old-school:
```
for (var checkDetail : checkDetails) {
    if (checkDetail.getBiometricVerificationProcessLevel() != null) {
        return checkDetail.getBiometricVerificationProcessLevel();
    }
}
return 0;
```

**Review by Joe-Edwards-GDS — DISMISSED** [review]
One optional comment, but looking good I think.

We should keep an eye out for the fact we've changed the validation rules for DCMAW, but it should be fine. (Maybe just wait until tomorrow morning to merge)

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**
